src/ScopeAsSelect.php

```diff
@@ -18,9 +19,34 @@ public static function builder($value): Builder
         return $value;
     }
 
+    public static function makeCallable($value): callable
+    {
+        $scopes = Arr::wrap($value);
+
+        return function ($query) use ($scopes) {
+            foreach ($scopes as $key => $scope) {
+                $arguments = [];
+
+                if (is_string($key)) {
```

Comment:
Refactor to `is_numeric` instead of `is_string`

https://github.com/laravel/framework/blob/7.x/src/Illuminate/Database/Eloquent/Builder.php#L1141